src/backend/executor/nodeSeqscan.c

```diff
@@ -379,3 +412,55 @@ ExecSeqScanInitializeWorker(SeqScanState *node,
 	node->ss.ss_currentScanDesc =
 		table_beginscan_parallel(node->ss.ss_currentRelation, pscan);
 }
+
+/*
+ * Returns true if the element may be in the bloom filter.
+ */
+static bool
+PassByBloomFilter(SeqScanState *node, TupleTableSlot *slot)
+{
+	Datum val;
+	bool isnull;
+
+	ListCell *lc;
+	foreach (lc, node->filters)
+	{
+		ScanKey sk = lfirst(lc);
+		if (sk->sk_flags != SK_BLOOM_FILTER)
+			continue;
+
+		val = slot_getattr(slot, sk->sk_attno, &isnull);
+		if (isnull)
+			continue;
+
+		bloom_filter *bf = (bloom_filter *)DatumGetPointer(sk->sk_argument);
+		if (bloom_lacks_element(bf, (unsigned char *)&val, sizeof(Datum)))
+			return false;
+	}
+
+	return true;
+}
+
+ScanKey
+ScanKeyListToArray(List *keys, int *num)
+{
+	ScanKey rs;
+
+	if (!num)
+		return NULL;
```

Comment:
When will `num` be NULL? It looks never in our kernel code.

Reply:
`change to Assert(num);`